Doc/howto upgrade (#129)
canonical/wazuh-server-operator: docs/how-to/upgrade.md

# How to perform a minor upgrade

**Example**: Wazuh 4.9 -> Wazuh 4.10.

```{important}

> Workload's versions are pinned in the charms. Not automatic upgrades will happen if not triggered at the charm level.

Reviewer: ```suggestion
The workload's versions are pinned in the charms. No automatic upgrades will happen if they're not triggered at the charm level.
```

> ```

Reviewer: Thanks for using admonitions! 

Nitpick for the GitHub render, since admonitions render as code blocks in GitHub: I would split the text across multiple lines so that everything can be read without needing to scroll across.

## Wazuh Server

> The Wazuh server has a stateless workload, and can safely be upgradaed through `juju refresh`.

Reviewer: ```suggestion
The Wazuh server has a stateless workload. It can safely be upgraded through `juju refresh`.
```

```{note}

> While the workload is stateless, some data are temporarily stored on disk before being sent to Wazuh Indexer. So you should not destroy/recreate the application and/or the units.

Reviewer: ```suggestion
While the workload is stateless, some data are temporarily stored on disk before being sent to Wazuh Indexer. So you should not destroy or recreate the application and/or the units.
```

> ```

Reviewer: same nitpick about splitting the text across multiple lines for easier readability on GitHub

# How to perform a minor upgrade

**Example**: Wazuh 4.9 -> Wazuh 4.10.

```{important}
The workload's versions are pinned in the charms.
No automatic upgrades will happen if they're not triggered at the charm level.
```

> ## Wazuh Server

Reviewer: ```suggestion
## Wazuh server
```

The Wazuh server has a stateless workload. It can safely be upgraded through `juju refresh`.

```{note}
While the workload is stateless, some data are temporarily stored on disk before being sent
to Wazuh Indexer. So you should not destroy or recreate the application and/or the units.
```

> ## Wazuh Dashboards

Reviewer: ```suggestion
## Wazuh dashboards
```

The Wazuh dashboards charm has a stateless workload. It can safely be upgraded through `juju refresh`.

> ## Wazuh Indexer

Reviewer: ```suggestion
## Wazuh indexer
```
